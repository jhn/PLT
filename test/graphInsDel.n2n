actor: Node = { name: String, age: Int };
actedIn: Rel = { role: String };
movie: Node = { title: String, year: Int };

fn main() -> Void {

	Keanu: Node = actor["Keanu", 35];
	Leo: Node = actor["Leo", 20];

	neo: Rel = actedIn["Neo"];
	jordan: Rel = actedIn["Jordan"];

	matrix: Node = movie["Matrix", 1999];
	wolf: Node = movie["Wolf", 1994];

	Cast: Graph = <
		Keanu neo matrix,
		Leo jordan wolf
	>;

	print(Cast);
	print("\n");
	print("\n");

	Cast ^+ (Keanu actedIn["Nelson"] movie["Sweet November", 2000]);
	print(Cast);
	print("\n");
	print("\n");

	Cast ^- (Keanu actedIn["Nelson"] movie["Sweet November", 2000]);
	print(Cast);
	print("\n");
	print("\n");

}
