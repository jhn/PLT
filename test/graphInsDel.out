( {Keanu, 35} {Neo} {Matrix, 1999},
  {Leo, 20} {Jordan} {Wolf, 1994}
)

( {Keanu, 35} {Neo} {Matrix, 1999},
  {Leo, 20} {Jordan} {Wolf, 1994},
  {Keanu, 35} {Nelson} {Sweet November, 2000}
)

( {Keanu, 35} {Neo} {Matrix, 1999},
  {Leo, 20} {Jordan} {Wolf, 1994}
)
